data/core/commands/doc.lua

```diff
@@ -120,6 +120,14 @@ local commands = {
     end
   end,
 
+  ["doc:type-unicode"] = function()
+    core.command_view:enter("Unicode code point", function(text)
+      -- It is possible both HEX and DEC, tonumber will converts
+      doc():text_input(common.codepoint_to_utf8(tonumber(text)))
+    end)
+    core.command_view:set_text("0x")
+  end,
```

Comment:
Ideally you'd want to also check for the validity of the inserted text. For example if you press enter without inserting anything, an error is thrown.

You could do that in the submit callback and/or in the validate callback.

Reply:
Check also if the number is negative (or decimal).